elastic/kibana
# [Fleet] Allow preconfigured output fields to be edited

## Summary

Resolve https://github.com/elastic/kibana/issues/156997
Related to https://github.com/elastic/ingest-dev/issues/1530

Allow to have preconfigured outputs with field editable in the UI.

For this I introduced a new property on preconfigured ouputs `allow_edit` that allow to specify a list of property that will be only populated by the preconfiguration value during the creation and will be editable in the UI and API.

## How to test 

You can create the following outputs in your kibana config file, and you should be able: 
* to create another default output
* Edit that output name

```
xpack.fleet.outputs:
  - id: test
    name: test output
    type: elasticsearch
    hosts: ['http://elasticsearch:9200']
    allow_edit: ['is_default', 'is_default_monitoring', 'name']
```

Example editing the name
<img width="767" alt="Screenshot 2023-05-15 at 12 01 16 PM" src="https://github.com/elastic/kibana/assets/1336873/5fbf3fb7-ab3e-4870-858d-f3735be488b4">
